os/pm/pm_activity.c

```diff
@@ -189,33 +96,28 @@ void pm_activity(int domain, int priority)
  *
  ****************************************************************************/
 
-void pm_stay(int domain, enum pm_state_e state)
+void pm_suspend(enum pm_domain_e domain)
 {
-	FAR struct pm_domain_s *pdom;
 	irqstate_t flags;
 
 	/* Get a convenience pointer to minimize all of the indexing */
 
 	DEBUGASSERT(domain >= 0 && domain < CONFIG_PM_NDOMAINS);
-	pdom = &g_pmglobals.domain[domain];
-
 	flags = enter_critical_section();
-	DEBUGASSERT(state < PM_COUNT);
-	DEBUGASSERT(pdom->stay[state] < UINT16_MAX);
-	pdom->stay[state]++;
+	DEBUGASSERT(g_pmglobals.suspend_count[domain] < UINT16_MAX);
```

Comment:
Could you change DEBUGASSERT to return ERROR and set_errno()?

If we control with each domain, we can return ERROR when we use api incorrectly.